### PR TITLE
Pypi prep

### PR DESCRIPTION
**Pull request by mbolivar**
A couple of patches I would like to merge before tagging v0.1.0 and uploading to PyPI.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,81 +1,99 @@
...
+To install everything in development mode instead, clone the West
+repository and run this from the top level directory::
+
+  pip3 install -e .
+
+If you don't want to change your system outside of cloning this
+repository, you can run the ``west`` package as a module::
 
-To run the test suite, use:
+  PYTHONPATH=/path/to/west/repository/src python3 -m west
```
Comment (carlescufi) [inline README.rst]: I guess this only works on *nix. On Windows one would have to resort to things like [these](https://superuser.com/a/1049433). In particular the easiest is probaby `cmd /C "set PYTHONPATH=/path/to/west/repo && python3 -m west"`

File: README.rst [hunk 2]
```diff
@@ -1,81 +1,98 @@
...
-----------
+  west init -w https://example.com/your-west-repository.git
+
+You can also add ``--west-rev some-branch`` to use ``some-branch``
+instead of ``master``.
+
+To use another manifest repository (optionally with ``--mr
+some-manifest-branch``)::
+
+  west init -u https://example.com/your-manifest-repository.git
```
Comment (carlescufi) [inline README.rst]: This made me realize that a common usecase would be to specify both a non-default west location and also a non-default manifest repo. Maybe we could add a combined option like:
`west init -uw https://github.com/myuser`, and west would automatically append `/west.git` and `/manifest.git` to the URL in order to fetch those.

Comment (mbolivar) [reply README.rst]: Eew :).

Sure, we could do something like that. Maybe `-b` or `--url-base`? I'm assuming this is post-0.1.0.

File: README.rst [hunk 3]
```diff
@@ -1,81 +1,98 @@
...
-Please use pip to install in development mode as documented below.
+  pip3 install west
 
-Important Note
---------------
+Then install the rest of West and a Zephyr development environment in
+a directory of your choosing::
 
-West is distributed in two pieces:
+  mkdir zephyr-dir && cd zephyr-dir
```
[the remark this answers is not in the document]
Comment (mbolivar) [reply README.rst]: We could, for sure, but I didn't want this to be ambiguous with the actual Zephyr tree that gets cloned by `west fetch`.

Do you want me to make this change?

[the remark this answers is not in the document]
Comment (carlescufi) [reply README.rst]: Right, in that case how about `mkdir zephyrproject && cd zephyrproject`?

File: README.rst (hunk 2 above)
Comment (carlescufi) [reply README.rst]: >Eew :)
You mean eew `-uw`? `-b` is fine with me :)

File: README.rst [hunk 4]
```diff
@@ -1,81 +1,98 @@
-This package contains the Zephyr RTOS meta tool, 'west'.
+This is the Zephyr RTOS meta tool, ``west``.
 
-WARNING
--------
+Installation
+------------
 
-DO NOT INSTALL WITH "python3 setup.py install".
+Install West's bootstrapper with pip::
```
Comment (carlescufi) [inline README.rst]: I wonder if we should capitalize West, or leave it lowercase `west`.

File: README.rst [hunk 5]
```diff
@@ -1,81 +1,98 @@
...
+- ``west init`` runs the bootstrapper, which clones the West source
+  repository and a *West manifest* repository. The manifest contains a
+  YAML description of the Zephyr installation, including Git
+  repositories and other metadata.
 
-- One of West's jobs is to manage interaction with Zephyr's multiple
-  Git repositories, including its own.
-- West is in its experimental stages and is moving quickly, meaning
-  users need to stay on HEAD.
+- ``west fetch`` clones the repositories in the manifest, creating
```
Comment (carlescufi) [inline README.rst]: maybe comment here that `west fetch` is run by the freshly cloned `west` code

File: README.rst [hunk 6]
```diff
@@ -1,81 +1,98 @@
...
-Installation
-------------
+Usage
+-----
 
-To install the West bootstrapping/wrapper script from this package in
-development mode, clone this repository and run this from the top
-level directory:
+West has multiple sub-commands. After running ``west init``, you can
+run them from anywhere under ``zephyr-dir``.
```
Comment (carlescufi) [inline README.rst]: `zephyrproject`

File: README.rst [hunk 7]
```diff
@@ -1,81 +1,98 @@
...
-WARNING
--------
+Installation
+------------
 
-DO NOT INSTALL WITH "python3 setup.py install".
+Install West's bootstrapper with pip::
 
-Please use pip to install in development mode as documented below.
+  pip3 install west
```
Comment (carlescufi) [inline README.rst]: One of the things I feel are extremely useful and sorely miss on Windows cmd line is git bash autocomplete. I wonder if we could reuse their script somehow and actually install it during bootstrapping.

File: README.rst [hunk 8]
```diff
@@ -1,81 +1,98 @@
...
+
+  west init -u https://example.com/your-manifest-repository.git
 
-To run the test suite, use:
+After ``init`` time, you can hack on the West tree in ``zephyr-dir``.
+
+To install everything in development mode instead, clone the West
+repository and run this from the top level directory::
+
+  pip3 install -e .
```
Comment (ulfalizer) [inline README.rst]: Would be nice to have a note on how you run `west` after running this command (`python -m west`). A plain `west` doesn't work.

File: README.rst (hunk 2 above)
Comment (mbolivar) [reply README.rst]: Oh, just seems a bit weird to me personally, but I see the value for short command lines.

File: README.rst (hunk 4 above)
Comment (mbolivar) [reply README.rst]: Sure, I'll lowercase it.

File: README.rst (hunk 5 above)
Comment (mbolivar) [reply README.rst]: I'm not sure what this means ... ``west fetch`` isn't run by the code, not sure how code can run something. Sorry to be dense, could you please clarify?

File: README.rst (hunk 7 above)
Comment (mbolivar) [reply README.rst]: Taking a serious look at autocomplete on all the platforms is on my wishlist as well -- should we start backlogging this in the issues for this repo?

File: README.rst (hunk 8 above)
Comment (mbolivar) [reply README.rst]: It should... what goes wrong?

**Comment by mbolivar** [comment]
Most changes addressed, some questions still remain though, @ulfalizer @carlescufi

File: README.rst (hunk 5 above)
Comment (carlescufi) [reply README.rst]: Just after installing `west` from PyPi with `pip`, you can run `west fetch`. The Python script that actually executes all the required `git fetch` operations is the one in the `west/` folder, not the one installed by pip, isn't it? This is no big deal, just a small clarification that might or might not be applicable.

File: README.rst (hunk 2 above)
Comment (carlescufi) [reply README.rst]: I think it makes sense, but this can wait if you don't want to add it now.

File: README.rst (hunk 7 above)
Comment (carlescufi) [reply README.rst]: Yes actually. I will create one right now.

File: README.rst (hunk 8 above)
Comment (ulfalizer) [reply README.rst]: Never mind. It's working.

Maybe a note could be added about needing `.local/bin/` in PATH.

**Comment by mbolivar** [comment]
> Approving as-is, although there are improvements that can be postponed. For now the biggest priority is to get this up on PyPi.

I've added a --base-url option to the bootstrap code and addressed README comments.

Should be good to go -- @ulfalizer?

**Comment by mbolivar** [comment]
> Should be good to go

I've also added a Python 2 version check and reordered the commits so the one resetting to 0.1.0 can be tagged properly.

**Comment by carlescufi** [comment]
Is this ready for merging now?

**Comment by mbolivar** [comment]
> do a point release when that's done

Sorry, I am wrong, no need to do another release for self-update, as that's non-bootstrap and won't get sent to pypi.

Moving west and the manifest would require changes to what's in pypi if we do it after making the first 0.1.0 upload.

File: README.rst (hunk 5 above)
Comment (carlescufi) [reply README.rst]: Nice, thank you

**Comment by carlescufi** [comment]
@ulfalizer please re-review so we can merge this and then rebase your PR on top

**Comment by carlescufi** [comment]
>The change to having west and manifest under a single top level directory would be pretty significant as well, since that would break existing installations and affect the bootstrapper. We should decide on that before asking for wider use.

That one is part of the bootstrap but in a separate PR, I suggest that we merge this first then rebase that PR on top.

As per publishing to pypi, I would publish as soon as this one is merged just because I'm nervous someone may take the `west` name otherwise.

**Comment by mbolivar** [comment]
> That one is part of the bootstrap but in a separate PR, I suggest that we merge this first then rebase that PR on top.
> 
> As per publishing to pypi, I would publish as soon as this one is merged just because I'm nervous someone may take the west name otherwise

+1 to both.

File: README.rst [hunk 9]
```diff
@@ -1,81 +1,110 @@
...
+
+After ``init`` time, you can hack on the west tree in ``zephyrproject``.
+
+To install everything in development mode instead, clone the west
+repository and run this from the top level directory::
+
+  pip3 install -e .
+
+(On Linux, make sure ``~/.local/bin`` is in your ``PATH`` if you go
+this route.)
```
Comment (ulfalizer) [inline README.rst]: Maybe this could be added at the end, within the parenthesis: "Alternatively, you can run ``west`` with ``python -m west``."

**Pull request merged**